src/Hint/List.hs

```diff
@@ -132,7 +132,10 @@ moveGuardsForward = reverse . f [] . reverse
               span (if any hasPFieldsDotDot (universeBi x)
                        || any isPFieldWildcard (universeBi x)
                       then const False
-                      else \x -> pvars' p `disjoint` vars_ x) guards
+                      else \x ->
+                       let pvars = pvars' p in
+                       pvars `disjoint` vars_ x && "pun-right-hand-side" `notElem` pvars
```

Comment:
What is "pun-right-hand-side"?

Reply:
See this code from `RdrHsSyn.hs`:
```
plus_RDR, pun_RDR :: RdrName
plus_RDR = mkUnqual varName (fsLit "+") -- Hack
pun_RDR  = mkUnqual varName (fsLit "pun-right-hand-side")
```
It comes up in the `pvar'` on this test case
```
issue619 = [pkgJobs | Pkg{pkgGpd, pkgJobs} <- pkgs, not $ null $ C.condTestSuites pkgGpd]
```

Reply:
That is an amazing hack in the GHC code base. Yuk! But fair enough, makes sense then - can you add a comment just above saying where that magic string is coming from. Any chance we can extract it from placeHolderPunRhs rather than hardcoding it?

Reply:
I put a comment for now. Not following what you are saying yet in 
```
Any chance we can extract it from placeHolderPunRhs rather than hardcoding it?
```

Reply:
It's fine with the comment. I can't believe GHC went there...